pkg/process/checks/process.go

```diff
@@ -106,20 +109,19 @@ func (p *ProcessCheck) Init(syscfg *SysProbeConfig, info *HostInfo) error {
 	}
 	p.networkID = networkID
 
-	p.maxBatchSize = getMaxBatchSize()
-	p.maxBatchBytes = getMaxBatchBytes()
+	p.maxBatchSize = getMaxBatchSize(p.config)
+	p.maxBatchBytes = getMaxBatchBytes(p.config)
 
-	p.skipAmount = uint32(ddconfig.Datadog.GetInt32("process_config.process_discovery.hint_frequency"))
+	p.skipAmount = uint32(p.config.GetInt32("process_config.process_discovery.hint_frequency"))
 	if p.skipAmount == 0 {
 		log.Warnf("process_config.process_discovery.hint_frequency must be greater than 0. using default value %d",
 			ddconfig.DefaultProcessDiscoveryHintFrequency)
-		ddconfig.Datadog.Set("process_config.process_discovery.hint_frequency", ddconfig.DefaultProcessDiscoveryHintFrequency)
```

Comment:
Why did we remove this? This was added so that if the config was set to 0 (a bit of an edge case), then we instead set the in-memory config value to the default value, to avoid hitting divide by zero errors: https://github.com/DataDog/datadog-agent/pull/14759/files#r1072614543

Reply:
There is actually a very good reason for this! Basically, the config component only provides the methods required to fulfill the `ConfigReader`, not `ConfigReaderWriter` (found [here](https://github.com/DataDog/datadog-agent/blob/79844b61deb3bb765da69546022dd5b7cb30a98e/pkg/config/types.go#L18-L68)). This has the nice effect of making the config immutable after it is loaded. Personally, I think this was the correct design decision because editing the config while the agent is running is a bit of an anti-pattern (with the notable exception of the runtime settings server) since it makes the config hard to reason about.

This was the only occurrence left of writing back values at runtime in the process-agent (once again, with the notable exception of the runtime setting server). If you look [below this line of code](https://github.com/DataDog/datadog-agent/blob/bfb19e80b58b6170e943fde94561aabc2fad27d8/pkg/process/checks/process.go#L116-L120), you can see that we actually do still validate the setting by resetting `p.skipAmount` to the default if the user sets it to be `0`, so there should still not be any divide by zero errors.

Sorry for the long-winded response. tldr - Calling `config.Set` is an anti-pattern, and we still do validation on this setting anyway.